wordpress-mobile/WordPress-Android
# [Reader] Connect fetch tags logic with compose UI

Fixes #20624 

https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/b5a79353-49f6-47b8-8366-5836f3388106

-----

## To Test:

- Install JP and sign in.
- Open "Reader".
- 🔍  Select "Tags" feed: loading, loaded, error and empty states should be displayed correctly ([actions not working yet](https://github.com/wordpress-mobile/WordPress-Android/issues/20670)).

-----

## Regression Notes

1. Potential unintended areas of impact

    - None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

3. What automated tests I added (or what prevented me from doing so)

    - Updated `ReaderTagsFeedViewModelTest.kt` and implemented `ReaderTagsFeedUiStateMapperTest.kt`.

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
